upt_macports/tests/test_macports_backend.py

```diff
@@ -64,6 +64,25 @@ def test_version_conversion(self):
             self.assertEqual(
                 self.macports_backend.cpan_version(mp_ver), cpan_ver)
 
+    @mock.patch('upt.Backend.needs_requirement')
+    def test_require(self, mock_need_req):
+        specifiers = {
+            '>=42': '>=42',
+            '<=42': '<=42',
+            '!=42': '!=42',
+            '==42': '==42',
+            '>= 1.2, != 1.5, < 2.0': '>=1.200.0, !=1.500.0, <2.0.0',
+            '>= 1.2, != 2, < 3.0': '>=1.200.0, !=2, <3.0.0'
+        }
+
+        for key, value in specifiers.items():
+            req = upt.PackageRequirement('bar', key)
+            self.macports_backend.needs_requirement(req, 'fake-phase')
+            self.assertCountEqual(
```

Comment:
so here you only check whether the number of elements is the same, correct? Can that be expanded to make sure that it actually gives the *exact* expected output as well, you have already the input (as ```key```) and expected output (as ```value```) listed in ```specifier```?

Reply:
At first even I thought that it just counts number of Elements as that is what the name 'assertCountEqual' suggest but actually it checks if the elements are equal or not too.

other option is assertListEqual but that needs the order of elements to be same too.

Reply:
ah, okay - sorry I should have read the documentation of the function ;)